numpy/random/_examples/numba/extending.py

```diff
@@ -5,80 +5,26 @@
 from timeit import timeit
 
 bit_gen = PCG64()
-next_d = bit_gen.cffi.next_double
-state_addr = bit_gen.cffi.state_address
 
-def normals(n, state):
-    out = np.empty(n)
-    for i in range((n + 1) // 2):
-        x1 = 2.0 * next_d(state) - 1.0
-        x2 = 2.0 * next_d(state) - 1.0
-        r2 = x1 * x1 + x2 * x2
-        while r2 >= 1.0 or r2 == 0.0:
-            x1 = 2.0 * next_d(state) - 1.0
-            x2 = 2.0 * next_d(state) - 1.0
-            r2 = x1 * x1 + x2 * x2
-        f = np.sqrt(-2.0 * np.log(r2) / r2)
-        out[2 * i] = f * x1
-        if 2 * i + 1 < n:
-            out[2 * i + 1] = f * x2
-    return out
+rg = np.random.Generator(PCG64())
 
-# Compile using Numba
-normalsj = nb.jit(normals, nopython=True)
-# Must use state address not state with numba
-n = 10000
 
-def numbacall():
-    return normalsj(n, state_addr)
+def numpycall(rg):
+    return rg.random(size=n)
 
-rg = np.random.Generator(PCG64())
 
-def numpycall():
-    return rg.normal(size=n)
+# Compile using Numba
+numbacall = nb.njit(numpycall)
+# Must use state address not state with numba
+n = 10000
 
 # Check that the functions work
-r1 = numbacall()
-r2 = numpycall()
+r1 = numbacall(rg)
+r2 = numpycall(rg)
 assert r1.shape == (n,)
 assert r1.shape == r2.shape
 
-t1 = timeit(numbacall, number=1000)
+t1 = timeit(lambda: numbacall(rg), number=n)
 print(f'{t1:.2f} secs for {n} PCG64 (Numba/PCG64) gaussian randoms')
-t2 = timeit(numpycall, number=1000)
+t2 = timeit(lambda: numpycall(rg), number=n)
 print(f'{t2:.2f} secs for {n} PCG64 (NumPy/PCG64) gaussian randoms')
-
-# example 2
-
-next_u32 = bit_gen.ctypes.next_uint32
-ctypes_state = bit_gen.ctypes.state
-
-@nb.jit(nopython=True)
-def bounded_uint(lb, ub, state):
```

Comment:
The same is true here as well. The method shown here is different from that in `Generator`.